test_common/harness/errorHelpers.cpp

```diff
@@ -748,10 +749,46 @@ const char * subtests_to_skip_with_offline_compiler[] = {
             "program_binary_type",
             "compile_and_link_status_options_log",
             "kernel_preprocessor_macros",
+            "execute_after_serialize_reload_library",
+            "execute_after_serialize_reload_object",
+            "execute_after_simple_compile_and_link",
+            "execute_after_simple_compile_and_link_no_device_info",
+            "execute_after_simple_library_with_link",
+            "execute_after_two_file_link",
+            "simple_compile_only",
+            "simple_compile_with_callback",
+            "simple_library_only",
+            "simple_library_with_callback",
+            "simple_library_with_link",
+            "simple_link_only",
+            "simple_link_with_callback",
+            "simple_static_compile_only",
+            "two_file_link",
 };
 
 int check_functions_for_offline_compiler(const char *subtestname, cl_device_id device)
 {
+    // We can't run any tests that make a call to clCompileProgram if we don't
+    // have a compiler for the device.
+    cl_bool is_device_compiler_available{false};
+    auto error = clGetDeviceInfo(device, CL_DEVICE_COMPILER_AVAILABLE,
+                                 sizeof(is_device_compiler_available),
+                                 &is_device_compiler_available, nullptr);
+    if (error != CL_SUCCESS)
+    {
+        log_error("ERROR: clGetDeviceInfo query CL_DEVICE_COMPILER_AVAILABLE "
+                  "failed: %s\n",
+                  error);
+        return 1;
+    }
+    if (!is_device_compiler_available) {
+        auto device_name = get_device_name(device);
+        log_info(
+            "Skiping subtest %s because there is no compiler for device: %s\n",
+            subtestname, device_name.c_str());
+        return 1;
```

Comment:
I'm not sure whether this should be considered a skip or a failure in the case that there is no compiler support?